src/Altinn.App.PlatformServices/Implementation/ApplicationLanguage.cs

```diff
@@ -0,0 +1,90 @@
+    {
+        private readonly AppSettings _settings;
+        private readonly ILogger _logger;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ApplicationLanguage"/> class.
+        /// </summary>
+        /// <param name="settings">The app repository settings.</param>
+        /// <param name="logger">A logger from the built in logger factory.</param>
+        public ApplicationLanguage(
+            IOptions<AppSettings> settings,
+            ILogger<AppResourcesSI> logger)
+        {
+            _settings = settings.Value;
+            _logger = logger;
+        }
+
+        /// <inheritdoc />
+        public async Task<List<Platform.Storage.Interface.Models.ApplicationLanguage>> GetApplicationLanguages()
+        {
+            string pathTextsResourceFolder = _settings.AppBasePath + _settings.ConfigurationFolder + _settings.TextFolder;
+
+            DirectoryInfo directoryInfo = new DirectoryInfo(pathTextsResourceFolder);
+
+            if (!directoryInfo.Exists)
+            {
+                _logger.LogWarning("The text resource directory does not exist");
+                return new List<Platform.Storage.Interface.Models.ApplicationLanguage>();
+            }
+
+            if (directoryInfo.GetFiles().Length < 1)
+            {
+                _logger.LogWarning("There are no resource files located in the text resource directory");
+                return new List<Platform.Storage.Interface.Models.ApplicationLanguage>();
+            }
+
+            var textResourceFilesInDirectory = directoryInfo.GetFiles();
+            var applicationLanguages = new List<Platform.Storage.Interface.Models.ApplicationLanguage>();
+
+            foreach (var fileInfo in textResourceFilesInDirectory)
+            {
+                try
+                {
+                    string fullFileName = Path.Join(pathTextsResourceFolder, fileInfo.Name);
+                    PathHelper.EnsureLegalPath(pathTextsResourceFolder, fullFileName);
+                    if (!File.Exists(fullFileName))
+                    {
+                        _logger.LogWarning("Something went wrong while trying to fetch the application language");
+                        return new List<Platform.Storage.Interface.Models.ApplicationLanguage>();
+                    }
+
+                    Platform.Storage.Interface.Models.ApplicationLanguage applicationLanguage;
+
+                    await using (FileStream fileStream = new(fullFileName, FileMode.Open, FileAccess.Read))
+                    {
+                        JsonSerializerOptions options = new() { PropertyNamingPolicy = JsonNamingPolicy.CamelCase };
+                        applicationLanguage = await System.Text.Json.JsonSerializer.DeserializeAsync<Platform.Storage.Interface.Models.ApplicationLanguage>(fileStream, options);
+                    }
+
+                    applicationLanguages.Add(applicationLanguage);
+                }
+                catch (Exception)
```

Comment:
No need for this "catch all" - just let the exception bubble up to the controller. In the controller you can choose to catch for example DirectoryNotFoundException and return a 404 to the client.

Reply:
I want to keep the "catch all" so I can log the warning, but then again I can rethrow the exception. What do you think @RonnyB71?

Reply:
If an exception is thrown an not caught it will be logged automatically, the warning log will only provide an explicit log message to search for and we will have both a Warning and an Error entry in the logs. If the extra logging provides additional information it makes sense to log it, but in this case it's just a static string.

In general I prefer to let exceptions bubble up unless you actually can handle it and do something about it. In this case we can't and it's really an exceptional state that an application doesn't have text files, so it's fine to fail.